event/event.go

```diff
@@ -0,0 +1,50 @@
+package event
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/bitrise-io/bitrise-step-analytics/models"
+)
+
+// Tracker ...
+type Tracker interface {
+	Send(analytics models.TrackEvent) error
+}
+
+type tracker struct {
+	topic   *pubsub.Topic
+	context *context.Context
+}
+
+func NewTracker(projectID string, topic string) Tracker {
+	ctx := context.Background()
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		panic(fmt.Sprintf("Couldn't start PubSub Client: %s", err.Error()))
+	}
+	return tracker{topic: client.Topic(topic), context: &ctx}
+}
+
+func (t tracker) Send(analytics models.TrackEvent) error {
+	properties := map[string]interface{}{"id": analytics.ID, "ts": convertEpochInMicrosecondsToBigQueryTimestampFormat(analytics.Timestamp), "event_name": analytics.EventName}
+	for k, v := range analytics.Properties {
+		properties[k] = v
+	}
+	payload, err := json.Marshal(properties)
+	if err != nil {
+		return err
+	}
+	_, err = t.topic.Publish(*t.context, &pubsub.Message{
```

Comment:
Should we introduce some retry mechanism? 🤔

Reply:
It seems Publish will not directly publish the event, but batch multiple and send them together.

To make this work properly we should call `Topic.Stop` at the end to force sending the remaining messages (according to the `Topic.Publish` comments), is it done?

Reply:
Regarding the retry, I think we should not wrap this block with a simple retry due to the above comment.

If we want probably we should create the Client with  `option.WithHTTPClient` and use our default retryable HTTP client.